doc/rbd/iscsi-target-ansible.rst

```diff
@@ -146,11 +103,16 @@ On the Ansible installer node, perform the following steps.
    ::
 
        # cd /usr/share/ceph-ansible
-       # ansible-playbook ceph-iscsi-gw.yml
+       # ansible-playbook site.yml --limit iscsigws
 
    .. note::
-    The Ansible playbook will handle RPM dependencies, RBD creation
-    and Linux IO configuration.
+    The Ansible playbook will handle RPM dependencies, setting up daemons,
+    and installing gwcli so it can be used to create iSCSI targets and export
+    RBD images as LUs. In past versions, iscsigws.yml could define the iSCSI
+    target and other objects like clients, images and LUNs, but this is no
+    longer supported. If iscsigws.yml has defintions for these object,
```

Comment:
Was this the solution to the backwards compatibility issue? I would have thought it would have just been "if ceph release < nautilus, do it the old way, else do it the new way"

Reply:
I let them mix and match if they really wanted to, because some users just did not use gwcli in the past and managed their iscsi/scsi objects like in Table 8.3. Operational Workflows here:

https://access.redhat.com/documentation/en-us/red_hat_ceph_storage/3/html/block_device_guide/using_an_iscsi_gateway#configuring_the_iscsi_target_using_ansible

where they updated the iscsigws.yml file to add/remove something like a client and they then reran the playbook.

So, I thought the user might want to update core ceph, but had to use the old ceph iscsi config because their scripts/tools relied on managing the cluster like described in that doc.

Reply:
At some point, we will need to deprecate ceph-iscsi-config from ceph-ansible. It might be good to just not advertise that the old way still works. Looking ahead to downstream products, we wouldn't want to support both under the same release -- we will want to kick them over to using gwlci and/or Ceph dashboard for management of LUNs.

Reply:
Ok I fixed this in the repush. You might have to repull again because the LUN fix and this one were in different repushes.